### ISSUE TITLE
Allow configuring upper/lower case on java Enum parsing and serialization

### ISSUE THREAD

**Issue by frankgrimes97** [plokhotnyuk/jsoniter-scala#446]
e.g. 
`public enum MyEnumType {ONE, TWO}`
{"key":"one"}

Not too familiar with the codebase, but I suppose this line could be tweaked based on some configs?
https://github.com/plokhotnyuk/jsoniter-scala/blob/master/jsoniter-scala-macros/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/macros/JsonCodecMaker.scala#L540

FYI Jackson supports this to some extent through:
https://fasterxml.github.io/jackson-databind/javadoc/2.10/com/fasterxml/jackson/databind/MapperFeature.html#ACCEPT_CASE_INSENSITIVE_ENUMS

**Comment by plokhotnyuk** [comment]
@frankgrimes97 Thanks for investigation and requesting an enhancement!

Currently W/A can be adding a custom codec for your Java enum like in the code snippet below:
```
import com.github.plokhotnyuk.jsoniter_scala.macros._
import com.github.plokhotnyuk.jsoniter_scala.core._

import scala.annotation.switch

case class MyCaseClassType(key: MyEnumType)

implicit val myEnumTypeCodec: JsonValueCodec[MyEnumType] =
  new JsonValueCodec[MyEnumType] {
    override def decodeValue(in:  JsonReader, default:  MyEnumType): MyEnumType =
      if (in.isNextToken('\"')) {
        in.rollbackToken()
        val l = in.readStringAsCharBuf();
        if (in.isCharBufEqualsTo(l, "one")) MyEnumType.ONE
        else if (in.isCharBufEqualsTo(l, "two")) MyEnumType.TWO
        else in.enumValueError(l)
      } else in.readNullOrTokenError(default, '\"')

    override def encodeValue(x: MyEnumType, out: JsonWriter): Unit =
      out.writeNonEscapedAsciiVal((x.ordinal(): @switch) match {
        case 0 => "one"
        case 1 => "two"
        case _ => out.encodeError(s"unexpected enum value: $x")
      })

    override val nullValue: MyEnumType = null
}

implicit val myCaseClassTypeCodec: JsonValueCodec[MyCaseClassType] =
  JsonCodecMaker.make(CodecMakerConfig)

val obj = readFromArray[MyCaseClassType]("""{"key":"one"}""".getBytes("UTF-8"))
val json = writeToArray(MyCaseClassType(key = MyEnumType.TWO))

println(obj)
println(new String(json, "UTF-8"))
```
It should print:
```
MyCaseClassType(ONE)
{"key":"two"}
```

**Comment by frankgrimes97** [comment]
Thanks, but we use many enums so overriding their encoding/decoding one by one in that way is quite verbose and tedious.

**Comment by plokhotnyuk** [comment]
@frankgrimes97 Please build a snapshot using the `sbt publishLocal` command from the corresponding branch and try if it will be suitable solution for your enums: https://github.com/plokhotnyuk/jsoniter-scala/pull/447

**Comment by frankgrimes97** [comment]
Hi, I'm getting errors when trying to build:
It looks like it can't find some plugins dependencies.
Any hints?

```
[info] Resolved  dependencies
[warn] 
[warn] 	Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn] 		com.typesafe:sbt-mima-plugin:0.6.1 (sbtVersion=1.0, scalaVersion=2.12)
[warn] 		com.github.gseitz:sbt-release:1.0.12 (sbtVersion=1.0, scalaVersion=2.12)
[warn] 		com.jsuereth:sbt-pgp:2.0.1 (sbtVersion=1.0, scalaVersion=2.12)
```

```
$ sbt --version
sbt version in this project: 1.3.5
sbt script version: 1.3.5
```

**Comment by plokhotnyuk** [comment]
@frankgrimes97 I've released the `2.1.0` version, artifacts are available from the Maven Central repo:
repo1.maven.org/maven2/com/github/plokhotnyuk/jsoniter-scala/jsoniter-scala-macros_2.12/2.1.0/
